drivers/interrupt_controller/intc_ite_it8xxx2.c

```diff
@@ -173,9 +173,16 @@ void ite_intc_irq_handler(const void *arg)
 uint8_t get_irq(void *arg)
 {
 	ARG_UNUSED(arg);
-	intc_irq = IVECT - IVECT_OFFSET_WITH_IRQ;
 
+	/* wait until two equal interrupt values are read */
+	do {
+		intc_irq = IVECT;
```

Comment:
Is it correct to write this register in the loop?  Isn't one write the correct answer so you don't wipe out the generated value in that small window from the condition to the reassignment?

Reply:
Added comment.